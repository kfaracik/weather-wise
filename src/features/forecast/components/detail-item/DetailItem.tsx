import React from 'react';
import {
  StyleSheet,
  Text,
  View,

  Image,
} from 'react-native';

export const DetailItem = ({iconSource, label}: {iconSource: any; label: string}) => (
  <View style={styles.detailItem}>
    <Image source={iconSource} style={styles.icon} />
    <Text>{label}</Text>
  </View>
);

const styles = StyleSheet.create({
  detailItem: {
    gap: 8,
    flexDirection: 'row',
    alignItems: 'center',
  },
  icon: {
    width: 20,
    height: 20,
    resizeMode: 'contain',
  },
});
