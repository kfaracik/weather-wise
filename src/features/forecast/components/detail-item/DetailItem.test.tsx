import React from 'react';
import {render, fireEvent, waitFor} from '@testing-library/react-native';
import {DetailItem} from './DetailItem';

describe('LocationSearchScreen', () => {
  it('should generate snapshot', () => {
    const {toJSON} = render(<DetailItem iconSource='img_source' label={'abc123!@#'} />);
    expect(toJSON()).toMatchSnapshot();
  });
});
