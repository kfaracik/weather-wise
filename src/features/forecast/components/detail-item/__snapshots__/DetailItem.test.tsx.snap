// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocationSearchScreen should generate snapshot 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "gap": 8,
    }
  }
>
  <Image
    source="img_source"
    style={
      {
        "height": 20,
        "resizeMode": "contain",
        "width": 20,
      }
    }
  />
  <Text
    style={
      [
        {
          "textAlign": "left",
        },
        {
          "color": "rgba(28, 27, 31, 1)",
          "fontFamily": "System",
          "fontWeight": "400",
          "letterSpacing": 0,
        },
        {
          "writingDirection": "ltr",
        },
        undefined,
      ]
    }
  >
    abc123!@#
  </Text>
</View>
`;
